src/chainstate/stacks/db/blocks.rs

```diff
@@ -2743,6 +2761,7 @@ impl StacksChainState {
             }
         }
 
+        block_tx.commit().map_err(|e| Error::DBError(e))?;
```

Comment:
Why does the block_tx need to commit with this change? Or was it a bug that the block_tx was left uncommitted previously?

Reply:
It's a bug that it was not provided previously, but it didn't show up in any tests because we don't (yet) test for orphan cleanup.  Omitting this commit would prevent orphaned blocks -- i.e. blocks that we know for a fact can never be included in the chain state -- from being marked as such.

Reply:
I expanded the tests here to verify that invalid blocks will get orphaned.